### PR TITLE
feat(Masthead): Add slot to display client content in the masthead

### PR DESCRIPTION
**Pull request by markhigham**
## Related issue

#[issueid]

## Overview

Add an optional prop to masthead which allows the insertion of custom client jsx. 

## Reason

Vertical screen space is at a premium, especially on MODNET laptops. Sometimes it is a small as 570 available pixels.
This slot was added so that things like DatePickers can be inlined in the masthead to save space.

## Work carried out

- [x] New prop and slot in the masthead
- [x] Updated stories and tests

### CODE REVIEW

File: packages/react-component-library/src/components/TopLevelNavigation/Masthead/Masthead.test.tsx [hunk 1]
```diff
@@ -129,6 +129,15 @@ describe('Masthead', () => {
     })
   })
 
+  describe('custom client component', () => {
+    it('should render custom client component', () => {
+      wrapper = render(
+        <Masthead {...props} customClientComponent={<div>Hello world</div>} />
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/components/TopLevelNavigation/Masthead/Masthead.test.tsx:135]: Can we change the name of this prop to `rightSlot`?

File: packages/react-component-library/src/components/TopLevelNavigation/Masthead/Masthead.test.tsx [hunk 2]
```diff
@@ -129,6 +129,15 @@ describe('Masthead', () => {
...
+      )
+      expect(wrapper.getByText('Hello world')).toBeInTheDocument()
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/components/TopLevelNavigation/Masthead/Masthead.test.tsx:137]: 😉 

```suggestion
      expect(wrapper.getByText('Hello, World!')).toBeInTheDocument()
```

File: packages/react-component-library/src/components/TopLevelNavigation/Masthead/Masthead.tsx [hunk 3]
```diff
@@ -150,7 +155,7 @@ export const Masthead = ({
             {nav}
           </StyledInlineNav>
         ) : null}
-
+        {!!customClientComponent && customClientComponent}
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/components/TopLevelNavigation/Masthead/Masthead.tsx:158]:
```suggestion
        {customClientComponent}
```

**Review by m7kvqbe1 — COMMENTED** [review]
Does it need some styling to stop it sitting flush?

Screenshot 2024-09-17 at 14 04 49

File: packages/react-component-library/src/components/TopLevelNavigation/Masthead/Masthead.test.tsx (hunk 1 above)
Comment (markhigham) [reply packages/react-component-library/src/components/TopLevelNavigation/Masthead/Masthead.test.tsx:135]: I don't want to surround the slot with styling. It's an "advanced" feature so the end user should take responsibility for how it works?
There may also be an edge case where they want it flush

**Comment by markhigham** [comment]
Closing this PR as it is replaced by https://github.com/Royal-Navy/design-system/pull/3888
